hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/catalyst/catalog/HoodieCatalogTable.scala

```diff
@@ -129,7 +129,17 @@ class HoodieCatalogTable(val spark: SparkSession, var table: CatalogTable) exten
   /**
    * Table schema
    */
-  lazy val tableSchema: StructType = table.schema
+  lazy val tableSchema: StructType = if (table.schema.nonEmpty) {
+    table.schema
+  } else {
+    val schemaFromMetaOpt = loadTableSchemaByMetaClient()
```

Comment:
this already handled in `parseSchemaAndConfigs()` , isn't it?

Reply:
@jinxing64 can you also have a look pls? given this is based on your previous change

Reply:
> this already handled in `parseSchemaAndConfigs()` , isn't it?
yes